pkg/sync/v5/clean.go

```diff
@@ -99,6 +99,7 @@ func clean(o unstructured.Unstructured) error {
 	case "ImageStream.image.openshift.io":
 		jsonpath.MustCompile("$.metadata.annotations.'openshift.io/image.dockerRepositoryCheck'").Delete(o.Object)
 		jsonpath.MustCompile("$.spec.tags[*].generation").Delete(o.Object)
+		jsonpath.MustCompile("$.spec.tags.*.referencePolicy.type").DeleteIfMatch(o.Object, "")
```

Comment:
doesn't make sense: clean() is really about removing runtime state from a live server object in order to be able to compare goal states

Reply:
as discussed, please add a line or two of clarifying comment about purpose in clean.go and default.go